Form and UX

jovasal opened the pull request:
Build a form with the following requirements:

Make sure that:
Name field accepts up to 30 chars.
Email field accepts only emails.
Message field accepts up to 500 chars.
All fields have placeholder text.
All fields have the correct type.
A user cannot submit the form if one or more of the fields is empty.
Collect data by using the [Formspree service](https://formspree.io/html).
In order to lay out the elements on the page you should use flexbox.
You don't need to implement any functionality that requires JavaScript.
You must stick to the design as much as possible (e.g., font, colors, images, text, margins) using [the templates in Figma](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=0%3A1).
Implement the button interactions (enable, hover, etc.).
The website should be responsive, following these breakpoints:
Minimum mobile screen size should be 375px
Minimum desktop screen size should be 768px
Use responsive values in your CSS rules, like percentages (i.e. width: 80%), instead of pixels

jovasal committed:
linters done

ok

jovasal committed:
eschedule ok

schedule ok

jovasal committed:
ok

ok

jovasal committed:
fillin all portfolio

fillomg all portfolio

jovasal committed:
ok

ok

jovasal committed:
change

change

jovasal committed:
spaces

spaces

jovasal committed:
ok

ok

jovasal committed:
changes donde

changes done for linters errors

jovasal committed:
linters

linters

jovasal committed:
margin

marginn 0

jovasal committed:
done

done

Jtaborda21 commented on style.css:
```diff
@@ -0,0 +1,1781 @@
+#consection {
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  justify-content: space-between;
```
_Kindly remove one of this line. Unexpected duplicate "space-between"   declaration-block-no-duplicate-properties_

jovasal replied:
Thank you so much!!!

I delete that.

justify-content: space-between;

Now is ok.

Have fun!

Jtaborda21 reviewed (commented):
Hi,
There are some issues that you still need to work:

jovasal committed:
error fixed

Error fixed

jovasal committed:
Create .hintrc

jovasal committed:
Create .stylelintrc.json

jovasal committed:
pictures

ok

Zuhaib042 reviewed (commented):
Hi @jovasal @Jtaborda21 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is a pair-programming activity, but I cant see any commit from @Jtaborda21 .kindly consider getting an actual code commit from him not a small change in the readme file. 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

jovasal committed:
Changes required

Changes required

Jtaborda21 committed:
Update CSS

Efiamotu-1 commented on index.html:
```diff
@@ -0,0 +1,343 @@
+        </div>
+      </div>
+    </section>
+    <section id="consection">
```
- [x] Good job on completing the functionality of the form, however you have not completed the design part which have been pointed with the red circles and underline in the image below
- [x] Your fonts and color do not match what is on figma for your `contact me title`
- [x] Your paragraph as well does not match the figma provided design as shown below and it is not properly centered
- [x] The form fields for `name & email` do not have the appropriate border radius and they are not properly centered including the textare
- [x] Please improve on the styling for this section before resubmitting your project for a review :+1: :sparkles: 
- figma design
![fm](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/89055502/ff403553-9801-41bb-be9d-6fdf0818eea4)
- your design
![mf](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/89055502/9d9b7179-1754-41b0-a1f4-3e880cc2d880)

jovasal replied:
**Good job on completing the functionality of the form, however you have not completed the design part which have been pointed with the red circles and underline in the image below**:
The red circle is not a part of the design. Maybe somebody is working in this point now:
![image](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/35693330/1eed961d-9020-40fb-8776-a9d97e7d6c8d)

**Your fonts and color do not match what is on figma for your contact me title**: Contact Me is fixed now. Actually, I am using a family Poppins similar, because Poppins alone contains problems, so I am using: (font-family: "Poppins", sans-serif;)
![image](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/35693330/2e4bfcc8-11ef-4184-ad30-06ea285f1790)

**Your paragraph as well does not match the figma provided design as shown below and it is not properly centered**:
All is fixed and in the center.
![image](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/35693330/5d1e3f0e-3ff9-4405-8adf-16192949d32b)

**The form fields for name & email do not have the appropriate border radius and they are not properly centered including the textare** Name & email fixed; Border radius fixed and in the center
![image](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/35693330/3bf122a4-e8b3-4b3e-b356-b61188aee9f9)

**Please improve on the styling for this section before resubmitting your project for a review 👍 ✨** As I mentioned before, I am using (font-family: "Poppins", sans-serif;) because Poppins alone cant use at this moment. It was recommended for another code reviwer.

Efiamotu-1 commented on index.html:
[OPTIONAL]
- [x] As shown from the image below your potfolio project design structure has been distorted please endeavour to fix this design flaw before resubmitting. :+1: 
 -Your designs
![ed](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/89055502/2020d20a-97c3-4697-b4f1-7ed15c16f887)
![od](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/89055502/d988910c-2222-4da3-b1fb-c1b8af3a00c9)

jovasal replied:
Thank you so much!!

![image](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/35693330/78cd2b1a-e930-4d72-ae4e-2f49d7341fb7)


I made this chanches because a coder review before can't see a borders radious, Now is clean.

Efiamotu-1 reviewed (changes requested):
Hi @jovasal , 

![](https://user-images.githubusercontent.com/89055502/221963929-dc172827-5c2c-411b-bf4d-923659f626e1.gif) 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [![@Efiamotu-1](https://img.shields.io/github/followers/Efiamotu-1?label=Habeeb-Efiamotu-Musa&style=social)](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [![@EFYAMOTU](https://img.shields.io/twitter/follow/omarramoun?label=Habeeb-Efiamotu-Musa&style=social)](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[![Habeeb Efiamotu Musa](https://img.shields.io/badge/linkedin-@Habeeb-crimson)](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [![Habeeb Efiamotu Musa ](https://img.shields.io/badge/slack-@Habeeb-brightgreen)](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

jovasal committed:
done

done

jovasal committed:
linters errors

linters errors

jovasal committed:
change

The form fields for name & email do not have the appropriate border radius and they are not properly centered including the textare

jovasal committed:
color

color

shahadat3669 commented on index.html:
```diff
@@ -0,0 +1,343 @@
+        </div>
+      </div>
+    </section>
+    <section id="consection">
+      <ul class="contain">
+        <li class="contact">
+          <h1 >Contact me</h1>
+          <p>
+            If you have an application you</br>
+            are interested in developing, a</br>
+            feature that you need built or a</br>
+            project that needs coding.</br>
+            I’d love to help with it!            
+          </p>
+          <form action="https://formspree.io/f/xayzkwne" class="form" method="post">
+            <li class="form-group">              
+              <input class="types"  type="text" id="name" placeholder="John Vasquez" maxlength="30" required/>
+            </li>
+            <li class="form-group">              
+              <input class="types" type="email" id="email" placeholder="jovasal@gmail.com" required/>
+            </li>
+            <li class="form-group"> 
+             <textarea name="message" id="message" cols="30" rows="10" placeholder="Write your message here" maxlength="500" required></textarea>
+            </li>
+            <li class="form-group">
+              <input  value="Get in touch" type="submit" class="btn-input">
+            </li>
+          </form>
+        </li>
+      </ul> 
+        <div class="border-line">
+        </div>
+      </section>
```
- [x] 👍 Well done on your progress so far! As mentioned by the previous reviewer, there are still some improvements needed to make your layout consistent with the design. Please carefully review the design and make the necessary updates to your element layout. Additionally, the paragraph in your section doesn't match the provided design and it's not properly centered.
- [ ] [Optional] Additionally, there is a small issue with the vertical scroll bar appearing on your webpage. According to the design, there should be no vertical scroll bar visible. This could be due to the width and spacing settings used. Please review and adjust the width and spacing of your elements to ensure that the content fits within the available space without triggering the appearance of a vertical scroll bar. By resolving this issue, you will achieve a closer match to the intended design and improve the overall user experience.

[Figma-portfolio.webm](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/55840999/df5f3faf-785d-48b3-99c8-b227a922c14d)


Before resubmitting your project for review, please work on refining the styling for this section. Pay attention to the details and ensure that your layout and paragraph are in line with the design. Keep up the good work! 👍✨

jovasal replied:
All changes were made satisfactory.

Also. All details are in the center.

![image](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/35693330/911d8c1c-4ef1-4b49-b1bf-ec836ac7324e)


![image](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/35693330/aa238788-1976-4ccb-aa41-1333aefd15ba)


![image](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/35693330/7103a213-08c5-41f5-b7b4-446c0f328315)

shahadat3669 commented on index.html:
```diff
@@ -0,0 +1,343 @@
+             <textarea name="message" id="message" cols="30" rows="10" placeholder="Write your message here" maxlength="500" required></textarea>
+            </li>
+            <li class="form-group">
+              <input  value="Get in touch" type="submit" class="btn-input">
```
[Optional] 👍 Great job on implementing default, hover, and active styles for your button! However, it appears that the styles you've applied are not consistent with the design. To ensure a cohesive and visually appealing interface, I recommend reviewing the design guidelines and updating your button styles accordingly.

Please pay attention to the following points:

- [x] [Optional] Make sure the colors used for the hover and active states match the design specifications. Consistency is key to creating a polished user interface. 🎨
- [x] [Optional] Check the design for any additional effects such as gradients, shadows, or transitions that need to be applied to the button. These subtle details can significantly enhance the visual appeal. ✨
![image](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/55840999/d888aa39-e5eb-4516-b701-c00ed3fd3e41)

By aligning your button styles with the design, you'll create a more cohesive and professional-looking user interface. Keep up the great work! 👍

jovasal replied:
colors were changed to Hoover color as mentioned before.

![image](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/35693330/8f352055-688f-43f4-89ea-79c28cf0f657)

![image](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/35693330/72fb4f56-8f86-4623-b2a6-339aa4257bc9)

shahadat3669 commented on index.html:
```diff
@@ -0,0 +1,343 @@
+            I’d love to help with it!            
+          </p>
+          <form action="https://formspree.io/f/xayzkwne" class="form" method="post">
+            <li class="form-group">              
+              <input class="types"  type="text" id="name" placeholder="John Vasquez" maxlength="30" required/>
+            </li>
+            <li class="form-group">              
+              <input class="types" type="email" id="email" placeholder="jovasal@gmail.com" required/>
+            </li>
+            <li class="form-group"> 
+             <textarea name="message" id="message" cols="30" rows="10" placeholder="Write your message here" maxlength="500" required></textarea>
+            </li>
+            <li class="form-group">
+              <input  value="Get in touch" type="submit" class="btn-input">
+            </li>
```
👍 Great to see that you made an effort to improve your input fields based on the previous reviewer's feedback. However, there are still a few adjustments that need to be made:

- [x] Please ensure that you use proper spacing in your input fields to match the design. Take a look at the design and add the appropriate spacing. 🙏
- [x] [Optional] In your placeholders, you are currently using "Lorem Ipsum," which is not meaningful. Please refer to the image provided and update the placeholders accordingly to provide more relevant information. 🙏
![Frame 40 (1)](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/55840999/5fbd7b56-17ae-4b69-8b43-b3d0f9f5f1fd)


Taking care of these details will enhance the overall user experience and align your input fields with the design. Keep up the good work! 👍

jovasal replied:
I added the spaces mentioned before. Also, I changed name and email as described in FIGMA

![image](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/35693330/31852a6a-a122-475a-a811-3ea1ed314ceb)

shahadat3669 reviewed (changes requested):
Hi @jovasal , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ No linter Errors

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- [Optional] You can consider updating your PR title. The title of your pull request lacks professionalism. It is essential to have a clear and meaningful title that accurately describes the purpose of the pull request. Consider using titles like "Create a Sidebar for Mobile Menu" or "Add Comments Section for User" that provide a concise summary of the changes made. :+1: Please change the PR title which describes what you did in this PR.
- I appreciate your efforts in improving your commits.:+1: It's encouraging to see your commitment to enhancing the commit history. However, there is still room for further improvement. Please try to enhance the clarity and organization of your commit history. :pray:

| Your | Professional |
| --- | --- |
| ![image](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/55840999/ded995e1-38f8-4f26-ae06-a6f4badb5793) | ![image](https://github.com/AndresEspi/hello-world/assets/55840999/9d1c46b3-6706-428c-8de6-b5b5e7ea968a) |


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

jovasal committed:
change requiered

last pull request

rica213 commented on style.css:
```diff
@@ -0,0 +1,1821 @@
+  align-content: stretch;
+  box-sizing: border-box;
+  width: 375px;
+  height: 6000px;
```
- [ ] [OPTIONAL] Kindly set your height property to **auto** instead and add a **padding-bottom** to have vertical spacing at the bottom 😁 It would render your design more flexible ❤️ 

Your current design 👇🏽 👇🏽 

![Screenshot from 2023-06-08 07-10-42](https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/10439283/fa8fe271-af8f-4d2f-a495-5af057c6041f)

rica213 commented on index.html:
```diff
@@ -0,0 +1,343 @@
+        <div class="border-line">
+        </div>
+      </section>
+  </body>
```
- [ ] [OPTIONAL] Please add the horizontal line at the bottom of the page to match the design from Figma 👀 
It is important to have keen eyes for details, even the littlest detail might make a huge difference in user experience. This is the time to practice 🏃🏽 ❤️

rica213 reviewed (approved):
# 🟢 STATUS: APPROVED
Hi John
My name is Larissa. I am your code reviewer 🥰
Nice to see you again 🤗 

You have implemented all the required changes requested by the previous reviewers. 😍 
Your project is complete!  There is nothing else to say other than... it's time to merge it 🛸 🛸 
Congratulations! 🎉 

<img src="https://media2.giphy.com/media/26u4lOMA8JKSnL9Uk/giphy.gif"/>

## Highlights
- [ ] Great feedback in the comment section ❤️ 
- [ ] No Linters errors ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] Correct Github flow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

stanleySimeon commented on index.html:
Greeting!

You're doing great work so far, but there are some changes you'll need to fix to go to the next steps.

Kindly, I suggest you stick to the design as much as possible (e.g., font, colors, images, text, margins) using [the info of the templates in Figma](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=0%3A1).

> ## The Figma Design
>
><img width="219" alt="Screenshot 2023-06-07 at 5 53 25 PM" src="https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/91237525/70a1c1cb-802b-4902-8bf1-867031902a54">
>
> ## Your Design
>
><img width="1400" alt="Screenshot 2023-06-08 at 12 43 43 AM" src="https://github.com/jovasal/Complete-Mobile-Figma-Portfolio/assets/91237525/72518a9c-605b-46ba-8097-92c744a260c0">

stanleySimeon commented on .github/workflows/linters.yml:
Kindly, I suggest you update your Readme file to make it match the Microverse Readme template. 😄

stanleySimeon commented on style.css:
Great job so for, but I suggest you use media queries to make your design match the project requirements. 😄

stanleySimeon reviewed (changes requested):
# Hi @jovasal, @Jtaborda21, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

![](https://media2.giphy.com/media/vQGBDTkhtIunaJYmRF/giphy-downsized.gif?cid=29caca75sf3uq9y4qgulk5ybbojwg94b4zo6temfzeel417x&ep=v1_gifs_search&rid=giphy-downsized.gif&ct=g
)

## Highlights
- Correct use of Gitflow ✔️
- Linter checks are passed ✔️
- Good PR description ✔️
- Meaningful commit messages ✔️


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- updating your readme file
- adding your live demo link
- that your stick to the design as much as possible

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

jovasal committed:
changef to repositore

I am working in another repository, but accidentally, i sent this repository with other project.

jovasal committed:
linters issues fixed

spaces

jovasal committed:
linters errors

linters

jovasal committed:
linters

spaces

emmiiorji reviewed (commented):
Hi @jovasal, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

<div align='center'>
<img src='https://user-images.githubusercontent.com/20554583/220870678-4f45b7ed-1d79-4e6a-9338-da59224b2cbc.gif'/>
</div>

You have submitted a pull request for a different milestone for the **Portfolio: desktop version: day 2** activity. Kindly note that each milestone should be submitted in a different pull request.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

jovasal committed:
fixed

fixed

jovasal committed:
oko

okoo

jovasal committed:
ok

ok

jovasal committed:
don

done

jovasal committed:
done

done

jovasal committed:
change done

return yo figma mobile version

jovasal committed:
important changes to my mobile portfolio

Already done

jovasal committed:
changes requiered

requiered

jovasal committed:
changes done

done

jovasal committed:
finishin mobile portfolio

needs to check linters

jovasal committed:
fixed linters issues

linters

jovasal committed:
Update README.md
